src/puppeteer_live_view_server.js

```diff
@@ -0,0 +1,447 @@
+import http from 'http';
+import EventEmitter from 'events';
+import Promise from 'bluebird';
+import uuid from 'uuid/v4';
+import WebSocket from 'ws';
+import log from 'apify-shared/log';
+import { checkParamOrThrow } from 'apify-client/build/utils';
+import { createTimeoutPromise } from './utils';
+import { layout, indexPage, detailPage } from './puppeteer_live_view_client';
+
+const LOCAL_IPV6 = '::';
+const LOCAL_IPV4 = '127.0.0.1';
+const DEFAULT_SCREENSHOT_TIMEOUT = 3000;
+
+/**
+ * LiveViewBrowser encapsulates a Puppeteer's Browser instance and provides
+ * an API to capture screenshots and HTML of the Browser's Pages. It uses
+ * EventEmitter API to communicate relevant Page events.
+ *
+ * @param {Browser} browser A Puppeteer Browser instance.
+ * @param {String} [options.id] A unique ID of the LiveViewBrowser.
+ * @param {Number} [options.screenshotTimeout] Max time allowed for the screenshot taking process.
+ */
+export class PuppeteerLiveViewBrowser extends EventEmitter {
+    constructor(browser, opts = {}) {
```

Comment:
I'm not sure I follow. Do you mean we should just move the logic to private methods and then call those in the constructor? Because, unless I'm missing something, we need to set up those listeners for every created browser.